snyk/policy
# feat!: bundle library and emit types

#### What does this PR do?

This PR bundles the library using Vite into a package that contains the following files
- an ESM file: `./dist/index.js`
- a CJS file: `./dist/index.cjs`
- and the TS type declartions: `./dist/index.d.ts`

All nested packages have been rolled up into these top level files. Most of the contents of `index.ts` have been moved to `policy.ts` and now only the functions exported from `index.ts` are exposed as part of the public API. This better communicates what is public and what is not, as well as allowing us to make internal changes without the risk of breaking any clients that rely on nested packages.

Types are now also emitted for the first time!

#### How should this be manually tested?

Run `npm pack` to bundle the library and then `npm install` the resulting `snyk-policy-0.0.0.tgz` into Registry, CLI or any other consumer of the library to make sure that there are no breaking changes... besides updating/alignment of types.
